tests/scripts/all.sh

```diff
@@ -2083,6 +2083,18 @@ component_test_variable_ssl_in_out_buffer_len_CID () {
     tests/compat.sh
 }
 
+component_test_CID_no_debug() {
+    msg "build: Connection ID enabled, debug disabled"
+    scripts/config.py unset MBEDTLS_DEBUG_C
+    scripts/config.py set MBEDTLS_SSL_DTLS_CONNECTION_ID
+
+    CC=gcc cmake .
+    make
+
+    msg "test: Connection ID enabled, debug disabled"
+    make test
```

Comment:
Is this actually a non-regression test? In 2.28, the bug (which I'll fix in https://github.com/Mbed-TLS/mbedtls/pull/5730) isn't triggered by unit tests, only by `compat.sh`.

Reply:
It is a non-regression test in 3.x. Without the bug fix:
```
Test moving clients handshake to state: CLIENT_CHANGE_CIPHER_SPEC . Segmentation fault (core dumped)
```

On the other hand, in 2.28, `test_suite_ssl` passes in this configuration, even though the corresponding test exists. I don't know why. The level of support for CID looks the same in both branches.